man/systemd.special.xml

```diff
@@ -1167,6 +1167,52 @@
         </varlistentry>
       </variablelist>
     </refsect2>
+
+    <refsect2>
+      <title>Special User Slice Units</title>
+
+      <para>There are four <literal>.slice</literal> units which form the basis of the user hierarchy for
+      assignment of resources for user applications and services. See
+      <citerefentry><refentrytitle>systemd.slice</refentrytitle><manvolnum>7</manvolnum></citerefentry>
+      for details about slice units.</para>
+
+      <variablelist>
+        <varlistentry>
+          <term><filename>-.slice</filename></term>
+          <listitem>
+            <para>The root slice is the root of the user's slice hierarchy.
+            It usually does not contain units directly, but may be used to set defaults for the whole tree.</para>
+          </listitem>
+        </varlistentry>
+
+        <varlistentry>
+          <term><filename>app.slice</filename></term>
+          <listitem>
+            <para>By default, all user services and applications managed by
+            <command>systemd</command> are found in this slice.</para>
+          </listitem>
+        </varlistentry>
+
+        <varlistentry>
+          <term><filename>session.slice</filename></term>
+          <listitem>
+            <para>By default all essential session services and applications
+            are placed into <filename>session.slice</filename>.
+            The exact definition depends on the desktop environment</para>
```

Comment:
How is that expected to work? For example, dbus-daemon is usually an "essential" part of desktops, so should the maintainer of dbus-daemon be putting `Slice=session.slice` into `dbus.service`? Or is the maintainer of dbus-daemon meant to leave it as-is, and then GNOME, KDE, XFCE etc. are each expected to ship `.d` dropins that move it into `session.slice`?

(Or if you're using dbus-broker, the same questions, but for that.)

Reply:
I expected services to opt-in to this. For DBus it should be added to the service file.

Shipping drop-in configurations in each desktop would be quite a pain. And, one installed DE doing this would modify all other ones on the same machine (unless you do ugly shenanigans of creating drop-in configurations at login time). I find it acceptable to do this in the short term, but only if we can expect upstreams to adopt the setting.

I hope that in most cases it should be pretty obvious what should go into `session.slice` (and what is acceptable; e.g. many `gnome-settings-daemon` plugins are not important enough, but we'll probably just place them all into `session.slice` as they are tiny.)

Reply:
>I expected services to opt-in to this

That's also how I envisioned things, it's the main reason to upstream the slice definitions into systemd.

Reply:
If that's the case, it's probably a good idea to say so, so that user-service authors have a reference for what they should do. Something like:

> Essential session services should be configured to be part of this slice, by adding `Slice=session.slice` to their unit files.

As far as I can see, it's harmless for a service to be assigned to this slice (or to background.slice) in older versions of systemd that don't define it or do anything special to create it, for example if you have a new version of `dbus` running on an old version of `systemd`. Is that true?

Reply:
session.slice will be created implicitly 

It just won't have the nice pretty description or documentation link if someone introspects the slice

Reply:
I am changing it to (trying to integrate @smcv's suggestion with what was there):

```
            <para>All essential services and applications required for the
            session should use this slice.
            Such services should be configured to be part of this slice by
            adding <varname>Slice=session.slice</varname> to their unit files.</para>
```

Reply:
Nice.